src/libsyntax/ast.rs

```diff
@@ -454,12 +454,21 @@ pub enum NestedMetaItemKind {
     Literal(Lit),
 }
 
+/// A name of `MetaItem`.
+#[derive(Clone, Eq, RustcEncodable, RustcDecodable, Hash, Debug, PartialEq)]
+pub enum MetaItemName {
+    /// Unscoped name, e.g. `name` in `#[name(..)]`.
+    Single(Name),
+    /// Scoped name, e.g. `rustfmt::skip` in `#[rustfmt::skip]`.
+    Scoped(Vec<Name>),
```

Comment:
Could we use a `Path` here, rather than `Vec<Name>`?

Reply:
In fact I wonder if we could use `Path` instead of `MetaItemName` and always treat attributes as paths rather than names (we have to do this for macro attributes anyway)